app/models/name.rb

```diff
@@ -408,33 +408,95 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  include CreatedUpdatedScopes
 
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
   scope :not_deprecated, -> { where(deprecated: false) }
-
+  scope :with_description, -> { where.not(description_id: nil) }
+  scope :without_description, -> { where(description_id: nil) }
   ### Module Name::Spelling
   scope :with_correct_spelling, -> { where(correct_spelling_id: nil) }
-
-  # For a glitch discovered in the wild:
+  scope :with_incorrect_spelling, -> { where.not(correct_spelling_id: nil) }
   scope :with_self_referential_misspelling, lambda {
     where(Name[:correct_spelling_id].eq(Name[:id]))
   }
-
+  scope :with_synonyms, -> { where.not(synonym_id: nil) }
+  scope :without_synonyms, -> { where(synonym_id: nil) }
+  ### Module Name::Create
+  # Get list of Names that are potential matches when creating a new name.
+  # Takes results of Name.parse_name.  Used by NameController#create_name.
+  # Three cases:
+  #
+  #   1. group with author       - only accept exact matches
+  #   2. nongroup with author    - match names with correct author or no author
+  #   3. any name without author - ignore authors completely when matching names
+  #
+  # If the user provides an author, but the only match has no author, then we
+  # just need to add an author to the existing Name.  If the user didn't give
+  # an author, but there are matches with an author, then it already exists
+  # and we should just ignore the request.
+  #
+  scope :matching_desired_new_name, lambda { |parsed_name|
+    if parsed_name.rank == "Group"
+      where(search_name: parsed_name.search_name)
+    elsif parsed_name.author.empty?
+      where(text_name: parsed_name.text_name)
+    else
+      where(text_name: parsed_name.text_name).
+        where(author: [parsed_name.author, ""])
+    end
+  }
+  scope :matching_search_name,
+        ->(search_name) { where(search_name: search_name) }
   ### Module Name::Taxonomy
-  scope :with_classification_like,
+  scope :with_rank,
+        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  scope :with_rank_below,
+        ->(rank) { where(Name[:rank] < Name.ranks[rank]) }
+  scope :with_rank_classification_like,
```

Comment:
Sorry, I would like a more transparent name again, even if it is wordy.  "with_name_at_rank_in_classification"?  "with_rank_name_in_classification"?  No, that's also ambiguous, I'd rather "with_rank_and_name_in_classification".

Reply:
Don't apologize... This is exactly what i want to hear. `with_rank_and_name_in_classification` it is!